core/src/main/java/com/aem/demo/core/models/MainPageModel.java

```diff
@@ -34,24 +44,60 @@ public class MainPageModel {
     @Getter
     private List<NewsCardModel> models;
 
+    @Getter
+    List<Integer> pageNumbers;
+
+    List<NewsCardModel> holdModels;
+
     @PostConstruct
     private void init() {
         PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
         if (pageManager == null) {
             return;
         }
         models = new ArrayList<>();
+        holdModels = new ArrayList<>();
         Page page = pageManager.getPage(ROOT_PATH);
         Iterator<Page> pageIterator = page.listChildren();
         while (pageIterator.hasNext()) {
             Page currentPage = pageIterator.next();
             Resource jcrContent = currentPage.getContentResource();
-            models.add(jcrContent.adaptTo(NewsCardModel.class));
+            holdModels.add(jcrContent.adaptTo(NewsCardModel.class));
+        }
+
+        double listSize = holdModels.size();
```

Comment:
Yeah I see your point youre right but how is this working then yesterday we tried to add one extra page and it worked.
or is it because we did it manually thorugh CRXDE, I mean it did not come through RSS Feed

Reply:
Sorry, I missed that the `listSize` is alreardy `double`. Then the division works as expected! That's allright
But please notices that with the number of cards < 5 it shows an extra page.
So better use `Math.ceil()` anyway